packages/react-ui/src/widgets/legend/LegendWrapper.js

```diff
@@ -101,6 +114,10 @@ export default function LegendWrapper({
                     opacity={opacity}
                     onChangeOpacity={handleChangeOpacity}
                   />
+                  {selectedLayerOptions.map((key) => {
+                    const Component = customLayerOptions[key] || UnknownLegendOption;
+                    return <Component key={key} optionsKey={key} layerId={id} />;
```

Comment:
I think optionKey is a good addition, but in the task it says that the layer is passed as prop, not only the layer id.

Reply:
But passing the whole layer as a prop would mean more modifications to LegendWrapper. Meaning you have to pass to that component the whole layer of get the whole layer via a hook in that componente. Are you ok with that ?

Reply:
Modify LegendWrapper to receive `layerOptions: React.Fn[]` instead of `selectedLayerOptions` and `customLayerOptions`.
`layerOptions` will be:
```js
const layerOptions = layer.options.map((key) => {
                    const Component = customLayerOptions[key] || UnknownLegendOption;
                    return <Component key={key} optionsKey={key} layer={layer} />;
})
```
So you have the layer obj and you don't put that logic in LegendWrapper making it easier. In the LegendWrapper you only render the layerOptions in JSX and that's all.

Reply:
Okay so `layerOptions` will be passed to `LegendWrapper` kind of like a `children` prop

Reply:
Yep 👌🏻 I think that'd work